src/xcoobee/sdk/Bees.js

```diff
@@ -1,247 +1,254 @@
+      return beesPage;
+    };
+    const apiCfg = SdkUtils.resolveApiCfg(config, this._.config);
+    const params = { searchText };
+
+    return SdkUtils.startPaging(fetchPage, apiCfg, params);
+  }
+
+  /**
+   *
+   * @async
+   * @param {Array<Object<string, Object>>} bees - A mapping of bee names to bee parameters.
+   * @param {string} bees[].key - The bee name.  A 'transfer' bee will be ignored.
+   * @param {Object} bees[].value - The bee parameters.
+   * @param {Object} options - The bee take off options.
+   * @param {Object} options.process -
+   * @param {Array<{value: any, name: string}>} [options.custom] - custom properties
+   * @param {Array<email|XcooBeeId>} [options.process.destinations] -
+   * @param {string[]} options.process.fileNames -
+   * @param {string} [options.process.userReference] -
+   * @param {TransactionSubscriptions} [subscriptions]
+   * @param {Config} [config] - If specified, the configuration to use instead of the
+   *   default.
+   *
+   * @returns {Promise<SuccessResponse | ErrorResponse>} - The response.
+   * @property {number} code - The response status code.
+   * @property {Error} [error] - The response error if status is not successful.
+   * @property {string} [error.message] - The error message.
+   * @property {string} request_id - The ID of the request generated by the XcooBee
+   *   system.
+   * @property {Object} [result] - The result of the response if status is successful.
+   * @property {string} [result.ref_id] - A reference ID generated by the XcooBee
+   *   system.
+   *
+   * @throws {XcooBeeError}
+   */
+  async takeOff(bees, options, subscriptions, config = null) {
+    this._assertValidState();
+    const apiCfg = SdkUtils.resolveApiCfg(config, this._.config);
+    const { apiKey, apiSecret, apiUrlRoot } = apiCfg;
+
+    const directiveInput = {
+      filenames: options.process.fileNames,
+      user_reference: options.process.userReference || null,
+    };
+
+    if (subscriptions) {
+      // TODO: validate subscriptions
+      directiveInput.subscriptions = subscriptions;
+    }
+
+    if (
+      options.custom
+      && Array.isArray(options.custom)
+      && options.custom.length
+    ) {
+      directiveInput.custom = options.custom.slice();
+    }
+
+    if (
+      Array.isArray(options.process.destinations)
+      && options.process.destinations.length > 0
+    ) {
+      directiveInput.destinations = options.process.destinations.map(
+        (destination) => {
+          if (ApiUtils.appearsToBeAnEmailAddress(destination)) {
+            return { email: destination };
+          }
+          return { xcoobee_id: destination };
+        }
+      );
+    }
+
+    directiveInput.bees = Object
+      .keys(bees)
+      .map((beeName) => {
+        if (beeName !== 'transfer') {
+          return { bee_name: beeName, params: JSON.stringify(bees[beeName]) };
+        }
+        return null;
+      })
+      .filter(bee => bee);
```

Comment:
You can also pass the `Boolean` constructor in order to filter out the `null` instead of creating a new function for `filter`.
```
      .filter(Boolean);
```